### PR TITLE
[ENHANCEMENT] More tests for ReactiveThrottler

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java [hunk 1]
```diff
@@ -115,6 +118,82 @@ void throttleShouldRejectTasksWhenTheQueueIsFull() {
...
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+
+        Thread.sleep(500);
+        Mono.from(testee.throttle(Mono.fromRunnable(() -> executed.getAndSet(true)), NO_IMAP_MESSAGE)).block();
+        // And the task is not executed
+        assertThat(executed.get()).isTrue();
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java:145]:
```suggestion
        // And the task is executed
        assertThat(executed.get()).isTrue();
```

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java [hunk 2]
```diff
@@ -115,6 +118,82 @@ void throttleShouldRejectTasksWhenTheQueueIsFull() {
...
+            .threadCount(20)
+            .operationCount(5)
+            .runSuccessfullyWithin(Duration.ofSeconds(10));
+
+        Thread.sleep(100);
+
+        AtomicBoolean executed = new AtomicBoolean(false);
+        Mono.from(testee.throttle(Mono.fromRunnable(() -> executed.getAndSet(true)), NO_IMAP_MESSAGE)).block();
+        // And the task is not executed
+        assertThat(executed.get()).isTrue();
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java:195]:
```suggestion
        // And the task is executed
        assertThat(executed.get()).isTrue();
```

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java [hunk 3]
```diff
@@ -115,6 +118,82 @@ void throttleShouldRejectTasksWhenTheQueueIsFull() {
...
+        subscribe5.dispose();
+        subscribe6.dispose();
+        subscribe7.dispose();
+        subscribe8.dispose();
+
+        Thread.sleep(100);
+
+        Mono.from(testee.throttle(Mono.fromRunnable(() -> executed.getAndSet(true)), NO_IMAP_MESSAGE)).block();
+        // And the task is not executed
+        assertThat(executed.get()).isTrue();
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java:175]:
```suggestion
        // And the task is executed
        assertThat(executed.get()).isTrue();
```

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java [hunk 4]
```diff
@@ -115,6 +118,82 @@ void throttleShouldRejectTasksWhenTheQueueIsFull() {
...
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+        Mono.from(testee.throttle(Mono.delay(Duration.ofMillis(50)).then(), NO_IMAP_MESSAGE)).subscribe();
+
+        Thread.sleep(500);
+        Mono.from(testee.throttle(Mono.fromRunnable(() -> executed.getAndSet(true)), NO_IMAP_MESSAGE)).block();
+        // And the task is executed
+        assertThat(executed.get()).isTrue();
+    }
+    @Test
```
Comment (Arsnael) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java:147]: skip a line with previous test?

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java [hunk 5]
```diff
@@ -115,6 +118,82 @@ void throttleShouldRejectTasksWhenTheQueueIsFull() {
...
+        subscribe7.dispose();
+        subscribe8.dispose();
+
+        Thread.sleep(100);
+
+        Mono.from(testee.throttle(Mono.fromRunnable(() -> executed.getAndSet(true)), NO_IMAP_MESSAGE)).block();
+        // And the task is executed
+        assertThat(executed.get()).isTrue();
+    }
+    @RepeatedTest(10)
```
Comment (Arsnael) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/ReactiveThrottlerTest.java:177]: idem

**Review by Arsnael — APPROVED** [review]

**Pull request merged**